### PR TITLE
Theme dev

### PR DESCRIPTION
**Pull request by amcaplan**
### WHY are these changes introduced?

Makes some fixes to the `theme dev` command so it works as expected

### WHAT is this pull request doing?

1. Correcting flags
2. Using the generic passthrough flag parser for theme commands
3. Using generic `--path` flag for `dev` as well as replacing the same in `push` and `pull`
4. Adds a scope that was missing for theme devving

### How to test your changes?

`shopify theme dev --path=/path/to/your/theme`

### CODE REVIEW

**Review by karreiro — COMMENTED** [review]
Thank you, @amcaplan! I'm trying to 🎩, but I'm facing this error:

image

Am I missing something?

File: packages/theme/src/cli/commands/theme/dev.ts [hunk 1]
```diff
@@ -54,30 +50,13 @@ export default class Dev extends Command {
...
-    }
-    if (flags['live-reload']) {
-      command.push('--live-reload')
-      command.push(flags['live-reload'])
-    }
+    const store = getTheme(flags)
 
     const adminSession = await session.ensureAuthenticatedAdmin(store)
-    await execCLI2(command, {adminSession})
+    const storefrontToken = await session.ensureAuthenticatedStorefront(['devtools'])
```
[the remark this answers is not in the document]
Comment (amcaplan) [reply packages/theme/src/cli/commands/theme/dev.ts:59]: Ah, good point! I'll take that out.

**Comment by amcaplan** [comment]
@karreiro you're totally right, we were messing up on requesting scopes and passing tokens. I've updated the PR and made some updates on the 2.0 side as well to accept a separate storefront renderer token.

Until we release the next CLI2 version, we can use `SHOPIFY_CLI_2_0_DIRECTORY=/path/to/cli2 dev shopify theme ...` to test. That'll cause CLI3 to invoke the version of CLI2 in that directory.

**Comment by amcaplan** [comment]
Going to merge. Noting that this won't work properly until we bump CLI2 and update the version here as well. Probably early next week, in keeping with our usual Monday releases.

**Pull request merged**